File: app/src/test/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapperTest.java
package org.openfoodfacts.scanner.category.mapper;

import com.fasterxml.jackson.databind.ObjectMapper;

import junit.framework.Assert;

import org.junit.Test;

import java.io.IOException;
import java.util.List;

import org.openfoodfacts.scanner.category.model.Category;
import org.openfoodfacts.scanner.category.network.CategoryResponse;
import org.openfoodfacts.scanner.utils.FileHelper;

/**
 * Created by Abdelali Eramli on 01/01/2018.
 */

public class CategoryMapperTest {

    @Test
    public void fromNetwork_FullResponse_CategoryList() throws IOException {
        ObjectMapper mapper = new ObjectMapper();
        CategoryResponse response = mapper.readValue(FileHelper
                .readTextFileFromResources("mock_categories.json", this.getClass().getClassLoader()), CategoryResponse.class);
        List<Category> categories = new CategoryMapper().fromNetwork(response.getTags());
        Assert.assertEquals(response.getTags().size(), categories.size());
    }
}

File: app/src/test/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapperTest.kt
package openfoodfacts.github.scrachx.openfood.category.mapper

import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.google.common.truth.Truth.assertThat
import openfoodfacts.github.scrachx.openfood.category.network.CategoryResponse
import openfoodfacts.github.scrachx.openfood.utils.readTextFileFromResources
import org.junit.jupiter.api.Test
import java.io.IOException

/**
 * Created by Abdelali Eramli on 01/01/2018.
 */
class CategoryMapperTest {
    @Test
    @Throws(IOException::class)
    fun fromNetworkFullResponseCategoryList() {
        val response = jacksonObjectMapper().readValue(readTextFileFromResources("mock_categories.json", this.javaClass.classLoader!!), CategoryResponse::class.java)
        val categories = CategoryMapper().fromNetwork(response.tags)
        assertThat(response.tags).hasSize(categories.size)
    }
}